datafusion/core/src/dataframe.rs

```diff
@@ -162,9 +164,10 @@ impl DataFrame {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn unnest_column(self, column: &str) -> Result<DataFrame> {
+    /// [Unnest]: crate::logical_expr::Unnest
+    pub fn unnest_column(self, column: &str, preserve_nulls: bool) -> Result<DataFrame> {
```

Comment:
I think this is different than what @vincev proposed in  https://github.com/apache/arrow-datafusion/pull/7044#discussion_r1271325311

Specifically, there are three ways to make unnest plans:
1. `unnest` function
2. `Dataframe::unnest`
3. `LogicalPlanBuilder`

It seemed inconsistent to add a `preserve_nulls` argument in only one place

An alternative would be to add new `unest_with_options`, `DataFrame::unest_with_options` and `LogicalPlanBuilder::with_options` and leave the existing signatures alone.

What do reviewers think?

Reply:
🤔  seeing the scope of the required changes to add a new parameter, I am thinking `_with_options` would be preferable

Reply:
we could just keep the overload but add the options struct instead of only the bool flag.

Reply:
`_with_options` looks nice to me too, `DataFrame::unnest` would call `_with_options` preserving nulls so that users will keep getting the same behavior as before that is consistent with other data frame libraries and SQL unnest would use the DuckDB/ClickHouse behavior.

Reply:
Ok, I'll give it a try